exonum/src/runtime/dispatcher/mod.rs

```diff
@@ -418,19 +423,54 @@ impl Dispatcher {
         self.register_running_service(&instance);
         Ok(())
     }
+
+    // Try to configure the started instance of the service and catch panic if it occurs.
+    fn initialize_service(
+        &self,
+        runtime: &dyn Runtime,
+        fork: &Fork,
+        descriptor: InstanceDescriptor,
+        constructor: Vec<u8>,
+    ) -> Result<(), ExecutionError> {
+        let constructor_is_empty = constructor.is_empty();
+
+        let context = ExecutionContext {
+            interface_name: <dyn Initialize<Params = ()> as Interface>::INTERFACE_NAME,
+            ..ExecutionContext::new(self, fork, Caller::Blockchain {})
+        };
+        let call_info = CallInfo {
+            instance_id: descriptor.id,
+            method_id: INITIALIZE_METHOD_ID,
+        };
+        let args = constructor.into_bytes();
+
+        catch_panic(|| {
+            runtime
+                .execute(&context, &call_info, args.as_ref())
+                .or_else(|err| {
+                    // Default behavior for case if the service does not implement an `Initialize`
+                    // interface and constructor is empty.
+                    if constructor_is_empty && err.kind == Error::NoSuchInterface.into() {
```

Comment:
Why can't I initialize a service without any arguments?

Reply:
You can do it if you implement `Initialize` interface